models/mli/model_skopes_rules.py

```diff
@@ -0,0 +1,288 @@
+"""Skopes rules """
+
+import uuid
+import os
+import datatable as dt
+import numpy as np
+from h2oaicore.models import CustomModel
+from sklearn.preprocessing import LabelEncoder
+from h2oaicore.systemutils import physical_cores_count
+from h2oaicore.systemutils import temporary_files_path, remove, config
+from h2oaicore.systemutils import make_experiment_logger, loggerinfo, loggerwarning, loggerdebug
+
+
+class SKOPE_RULES(CustomModel):
+    _regression = False
+    _binary = True
+    _multiclass = False
+    _display_name = "SKOPE RULES"
+    _description = "SKOPE RULES"
+    _modules_needed_by_name = ['shap', 'collections', 'scipy', 'pandas', 'matplotlib', 'sklearn', 'skope-rules']
+
+    @staticmethod
+    def do_acceptance_test():
+        return True
+
+    def set_default_params(self, accuracy=None, time_tolerance=None,
+                           interpretability=None, **kwargs):
+        # Fill up parameters we care about
+        self.params = dict(random_state=kwargs.get("random_state", 1234),
+                           max_depth_duplication = None, n_estimators= 10,
+                           precision_min = 0.5, recall_min=0.01, max_samples = 0.8,
+                           max_samples_features = 1.0, max_depth = 3,
+                           max_features = "auto", min_samples_split = 2, 
+                           bootstrap= False, bootstrap_features = False)
+
+    def mutate_params(self, accuracy=10, **kwargs):
+        if accuracy > 8:
+            max_depth_duplication = [None, 2, 3]
+            n_estimators= [10, 20, 40]
+            precision_min = [0.1, 0.2, 0.3]
+            recall_min=[0.01, 0.05]
+            max_samples = [0.5, 0.8, 1.0]
+            max_samples_features = [0.5, 0.8, 1.0]
+            max_depth = [3, 4, 5]
+            max_features = ["sqrt", "log2", "auto"]
+            min_samples_split = [2, 11, 21]
+            bootstrap= [True, False]     
+            bootstrap_features = [True, False]  
+        elif accuracy >= 5:
+            max_depth_duplication = [None]
+            n_estimators= [10, 20]
+            precision_min = [0.1, 0.2, 0.3]
+            recall_min=[0.01]
+            max_samples = [0.8, 1.0]
+            max_samples_features = [1.0]
+            max_depth = [3, 4]
+            max_features = ["sqrt", "log2", "auto"]
+            min_samples_split = [2, 5, 11]
+            bootstrap= [True, False]     
+            bootstrap_features = [True, False]  
+        else:
+            max_depth_duplication = [None]
+            n_estimators= [10]
+            precision_min = [0.1, 0.2]
+            recall_min=[0.01]
+            max_samples = [0.8, 1.0]
+            max_samples_features = [0.8, 1.0]
+            max_depth = [3, 4]
+            max_features = ["auto"]
+            min_samples_split = [2]
+            bootstrap= [True, False]     
+            bootstrap_features = [True, False]  
```

Comment:
Is there reasoning behind parameters selected for these models? Maybe good to comment.